conda/recipes/ucxx/conda_build_config.yaml

```
@@ -15,4 +15,4 @@ python:
   - 3.10
 
 ucx:
-  - ">=1.13.0,<1.15.0"
+  - ">=1.13.0"
```

```suggestion
  - 1.13.0
```

---

Won't this be a tight pin though? What if we want to install 1.14.0, will that still work?

---

This only pins it during the build

As `ucx` has [`run_exports`]( https://github.com/conda-forge/ucx-split-feedstock/blob/8a97995a5cd61a740bbe5a82e55b946fc8399621/recipe/meta.yaml#L21-L22 ), this would become `ucx >=1.13.0,<1.14`

If we wanted to use `1.14.0` as the minimum, we could change this to `1.14.0` instead

---

Can we have a build that supports both `1.13.x` and `1.14.x` though? This is what I was trying to achieve.

---

We could turn this build into a matrix

```suggestion
  - 1.13.0
  - 1.14.0
```

The pinning itself comes from [the `run_exports`]( https://github.com/conda-forge/ucx-split-feedstock/blob/8a97995a5cd61a740bbe5a82e55b946fc8399621/recipe/meta.yaml#L21-L22 ). Related discussion ( https://github.com/conda-forge/ucx-split-feedstock/pull/92#discussion_r836882363 )

---

That would be better, I think. Would this mean we end up with two different package builds?

---

Yeah exactly

The challenge is the current constraint that is in `conda_build_config.yaml` doesn't do what we would want either. Conda would look at `ucx >=1.13.0,<1.15.0` in `host` and would pick `ucx 1.14.0` to match that. As a result the `run` line would become `ucx >=1.14.0,<1.15.0`. IOW not compatible with `1.13.x`

---

Does that mean that UCX-Py is also wrong? We have (as far as I understand) precisely the same we have here, [`ucx` in the `host` section](https://github.com/rapidsai/ucx-py/blob/7861326787abccb3cfb7b0f6da2234469cfc3248/conda/recipes/ucx-py/meta.yaml#L30) and pinning to [`>=1.13.0,<1.15.0`](https://github.com/rapidsai/ucx-py/blob/7861326787abccb3cfb7b0f6da2234469cfc3248/conda/recipes/ucx-py/conda_build_config.yaml#L7).